frontend/src/components/Pages/InspectionPage/InspectionTable.tsx

```diff
@@ -40,143 +45,196 @@ interface IProps {
-                        {YellowCircle} {TranslateText('Due this week')}
-                    </>
-                )
-            case deadlineDays > 7 && deadlineDays <= 14:
-                return (
-                    <>
-                        {YellowCircle} {TranslateText('Due within two weeks')}
-                    </>
-                )
-            case deadlineDays > 7 && deadlineDays <= 30:
-                return (
-                    <>
-                        {GreenCircle} {TranslateText('Due within a month')}
-                    </>
-                )
-        }
-        return (
+const getInspectionRow = (
+    inspection: Inspection,
+    ongoingMissions: OngoingMissionType,
+    openDialog: () => void,
+    setScheduledMissions: (selectedMissions: CondensedMissionDefinition[]) => void,
+    navigate: NavigateFunction
+) => {
+    const mission = inspection.missionDefinition
+    let status
+    let lastCompleted: string = ''
+    const isScheduled = Object.keys(ongoingMissions).includes(mission.id) && ongoingMissions[mission.id]
+    if (isScheduled) {
+        status = (
             <>
-                {GreenCircle} {TranslateText('Up to date')}
+                {GreenCircle} {TranslateTextWithContext('Already scheduled')}
             </>
         )
-    }
-
-    const getInspectionRow = (inspection: Inspection) => {
-        const mission = inspection.missionDefinition
-        let status
-        let lastCompleted: string = ''
-        const isScheduled = Object.keys(ongoingMissions).includes(mission.id) && ongoingMissions[mission.id]
-        if (isScheduled) {
-            status = (
-                <>
-                    {GreenCircle} {TranslateText('Already scheduled')}
-                </>
-            )
-        } else {
-            if (!mission.lastRun || !mission.lastRun.endTime) {
-                if (inspection.missionDefinition.inspectionFrequency) {
-                    status = (
-                        <>
-                            {RedCircle} {TranslateText('Not yet performed')}
-                        </>
-                    )
-                } else {
-                    status = TranslateText('No planned inspection')
-                }
-                lastCompleted = TranslateText('Never')
+    } else {
+        if (!mission.lastRun || !mission.lastRun.endTime) {
+            if (inspection.missionDefinition.inspectionFrequency) {
+                status = (
+                    <>
+                        {RedCircle} {TranslateTextWithContext('Not yet performed')}
+                    </>
+                )
             } else {
-                status = inspection.missionDefinition.inspectionFrequency
-                    ? getInspectionStatus(inspection.deadline!)
-                    : TranslateText('No planned inspection')
-                lastCompleted = formatDateString(mission.lastRun.endTime!)
+                status = TranslateTextWithContext('No planned inspection')
```

Comment:
Will this always be safe? Will getInspectionRow always be called the same number of times?

Reply:
Not sure what you mean by this? Can you clarify a bit, you're probably right

Reply:
I just mean that since TranslateTextWithContext calls a useContext hook, it needs to always be called in the same order at each page rerender. But since there are no errors at the moment then it's probably fine

Reply:
At the time I checked for errors with the hooks order and it looked fine but can try to pay a closer look